### PR TITLE
Feature/remove-skip-node-modules

### PR DESCRIPTION
**Pull request by AndreaPontrandolfo**

### CODE REVIEW

File: packages/eslint-config-sheriff/tsup.config.json [hunk 1]
```diff
@@ -8,6 +8,5 @@
   "splitting": false,
   "sourcemap": false,
   "clean": true,
-  "noExternal": ["@sherifforg/constants"],
-  "skipNodeModulesBundle": true
```
Comment (lishaduck) [inline packages/eslint-config-sheriff/tsup.config.json:12]: Does tsup work better now? I'm still concerned that tsup is handling this wrong.

Comment (AndreaPontrandolfo) [reply packages/eslint-config-sheriff/tsup.config.json:12]: im trying different stuff to solve the publishing error. Then running in different errors lol

Comment (AndreaPontrandolfo) [reply packages/eslint-config-sheriff/tsup.config.json:12]: @lishaduck after long exploring, apparently [this thing](https://github.com/egoist/tsup/issues/927#issuecomment-2416440833) is fixing all my issues, and i'm not sure why. But yeah, looks like an open bug on TSUP.

Comment (lishaduck) [reply packages/eslint-config-sheriff/tsup.config.json:12]: What publishing error? The dynamic require error is proof that it's not externalizing node_modules right, and the require thing is weird hack that works right, but means that you'll easily run into the dual package hazard.
The "correct" thing is not to bundle `node_modules`, as ESM code should never need a dynamic require.

File: packages/eslint-config-sheriff/src/index.ts [hunk 2]
```diff
@@ -9,18 +9,25 @@ import {
...
 export {
   getExportableConfig as sheriff,
   indexedBaseNoRestrictedSyntaxRules as baseNoRestrictedSyntaxRules,
-  allJsExtensions,
-  allJsxExtensions,
-  ignores,
-  sheriffStartingOptions,
-  supportedFileTypes,
-  testsFilePatterns,
+  exportableAllJsExtensions as allJsExtensions,
```
Comment (lishaduck) [inline packages/eslint-config-sheriff/src/index.ts:25]: What does this do? as but to constants? As usual, just a bit confused.

File: packages/eslint-config-sheriff/package.json [hunk 3]
```diff
@@ -46,8 +46,9 @@
   "scripts": {
     "preinstall": "npx only-allow pnpm",
     "clean": "rm -rf .turbo dist",
-    "build": "tsup",
-    "build-watch": "tsup",
+    "re-bundle-types": "dts-bundle-generator -out-file=dist/index.d.ts ./dist/index.d.ts",
```
Comment (lishaduck) [inline packages/eslint-config-sheriff/package.json:49]: Is this needed, or does dts.resolve resolve it? (If not, I might try experimentalDts, which should also work)

File: apps/sheriff-webservices/src/getAllRules.ts [hunk 4]
```diff
@@ -44,7 +44,6 @@ export const getAllRules = (
   const reactRulesCatalog = {
     ...react.configs.flat.recommended.rules,
     ...react.configs.flat['jsx-runtime'].rules,
-    // @ts-expect-error
```
Comment (lishaduck) [inline apps/sheriff-webservices/src/getAllRules.ts:47]: Yay! 🙃

**Review by lishaduck — COMMENTED** [review]
Just questions.
I think I'd be a lot less confused if I knew what was wrong.

**Pull request merged**